### PR TITLE
Homework 1 pull

### PR DESCRIPTION
**Pull request by PSUtblock**
Hey!

I don't have much to say other than it was a great refresher from over the break. Styling in particular flys out of my mind quickly if I do not use it often. The game was a great way to relearn some of that stuff. Though I will say I wonder how often I would use some of them. Nevertheless, fun. I did notice that in some instance you could ignore the use of * which i am uncertain if it is valid or not, or even necessary. 

As we were going some of the Javascript review in class I was just confirming my code. One thing I would say that I didn't do, but amended was making my function an expression rather than a declaration. This to me, is a bit annoying since I like my functions at the bottom of code. However, I learned that when you make a javascript function expression it isn't hoisted. which is rather annoying. In some ways I can see the benefit in the sense that the function expressions can only be used within their blocks and only be needed when they are needed, but to me that just seems disorganized. Though, in the case of making many objects in a single file I can see how that is just normal. 

I am sure I missed something, but for the most part I think it is a decent refresher!

solving the function I would check the palindrome on input (based from the templated script). Each time I checked if something was there and since we only wanted to positive integers I simply used math abs and floor to ensure that if you attempted to enter a number that was negative or a decimal, it would just. correct it. I like this implementation as it will always provide a response to the user. With the exception of using 'e' which I couldn't figure out with my knowledge.

once the number was entered I turned it into a string made a copy and reversed it to compare. then just printed the result into a div called 'response'.

As for the html, I just browsed around the bootstrap documentation and wanted saw the  'card' classes. I thought these looked cool and made a very unified look.

### CODE REVIEW

File: hw1/01-selectors.md [hunk 1]
```diff
@@ -1,7 +1,37 @@
...
+18. plate:nth-child(3)
+19. bento:nth-last-child(3)
+20. apple:first-of-type
+21. plate:nth-of-type(even)
+22. plate:nth-of-type(2n+3)
+23. plate apple:only-of-type
+24. orange:last-of-type, apple:last-of-type
+25. bento:empty
+26. apple:not(.small)
+27. *[for]
```
Comment (mlantas) [inline hw1/01-selectors.md:32]: I believe the star selector here is unnecessary. Removing it and leaving just `[for]` should accomplish the same result.

File: hw1/02-palindrome.html [hunk 2]
```diff
@@ -9,17 +9,18 @@
...
-  <body>
-    <main>
-      <section>
-        <h1>Exercise 02 - Palindrome</h1>
-        <div>
-          <label>Enter a positive number:</label>
-          <input type="number" />
+  <body class="bg-primary">
+    <main class="container-fluid">
+      <section class="m-2 card d-flex bg-light shadow">
```
Comment (mlantas) [inline hw1/02-palindrome.html:14]: With these bootstrap styles, the card stretches to fill the available width of the viewport: 
image

You should consider using one of the bootstrap display classes or another method to limit the width of the element.

File: hw1/02-palindrome.js [hunk 3]
```diff
@@ -1,3 +1,30 @@
 const elem = document.querySelector('input');
+const response = document.getElementById('response');
 
-elem.addEventListener('input', handleInput);
+//function that checks if the number entered is a palindrome
+const handleInput = function handleInput() {
+
+    //quick way to not allow for negative integers or decimal values
+    if(elem.value)
+        elem.value = Math.floor(Math.abs(elem.value));
```
Comment (mlantas) [inline hw1/02-palindrome.js:9]: This is a cool way to limit the number input! 

However, just from reading the code and comment, it was not immediately apparent to me that it is overriding the text inside the html input element on the actual webpage. I suggest tweaking the comment to mention that this modifies the value displayed to the user on the page as well.

**Review by mlantas — DISMISSED** [review]

File: hw1/02-palindrome.html [hunk 4]
```diff
@@ -9,17 +9,18 @@
...
   </head>
-  <body>
-    <main>
-      <section>
-        <h1>Exercise 02 - Palindrome</h1>
-        <div>
-          <label>Enter a positive number:</label>
-          <input type="number" />
+  <body class="bg-primary">
+    <main class="container-fluid">
```
Comment (golson-pdx) [inline hw1/02-palindrome.html:13]: Similar to what mnxn said, I think the container-fluid class here can be changed to be a smaller width or include a different element within that can be used to set maximum width. You could also limit width of all elements within container if you don't want to target the container itself.

**Review by golson-pdx — COMMENTED** [review]
Aside from some recommendations from mnxn not much else I saw that needs a lot of reworking, great job at following the Airbnb guide in your JS file!

File: hw1/02-palindrome.js (hunk 3 above)
Comment (PSUtblock) [reply hw1/02-palindrome.js:9]: Fair point.

**Comment by ellie9393** [comment]
Great job following airbnb style guide and using bootstrap classes. Everything looks good to me, the only thing I noticed was that when I ran your code and I was trying different inputs, I entered a number following a '.' and more numbers, and the numbers following the '.' were typed to the top! For example: I entered  "123.456", and in the input box, it typed "456123".
I tried to figure it out, but no luck so far.

**Pull request merged**